healthcheck.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+# Health check script for Vector
+# Returns 0 if healthy, non-zero if unhealthy
+# Detects when Vector is running with minimal/console-only configuration
+
+set -euo pipefail
+
+# Check if Vector is healthy (returns JSON: {"ok":true} or {"ok":false})
+VECTOR_HEALTH=$(curl -s http://localhost:8686/health 2>/dev/null | jq -r '.ok' 2>/dev/null || echo "error")
+
+if [ "$VECTOR_HEALTH" != "true" ]; then
+    echo "Vector health check failed - not responding or unhealthy ($VECTOR_HEALTH)"
+    exit 1
+fi
+
+# Check what sinks Vector has configured
+# If only console sink exists, something is wrong
+SINKS=$(curl -s http://localhost:8686/graphql \
+    -H "Content-Type: application/json" \
+    -d '{"query":"{ sinks { edges { node { componentId componentType } } } }"}' 2>/dev/null | \
+    jq -r '.data.sinks.edges[].node.componentId' 2>/dev/null || echo "")
+
+if [ -z "$SINKS" ]; then
+    echo "ERROR: Vector has no sinks configured"
+    exit 1
+fi
+
+# Count how many sinks we have
+SINK_COUNT=$(echo "$SINKS" | wc -l)
+
+# Check if only console sink exists (emergency/fallback mode)
+if [ "$SINK_COUNT" -eq 1 ] && echo "$SINKS" | grep -q "^console$"; then
+    echo "ERROR: Vector running with console-only sink (lost configuration)"
+    exit 1
+fi
```

Comment:
Is this going to crash loop collector if I use manual config with just one sink? Could we check for specific sink name?

Reply:
Even if you only have one sink in manual, the generated config has 3 sinks. Also, this line checks if that 1 sink is the default `console` sink. Assuming the generated config was picked up, this should never be 1.

Reply:
I was under impression that if you uncheck all checkboxes you get empty generated config.

Reply:
No, I just checked (unchecked everything in a Kubernetes collector) - we always build the Better Stack http sinks, with the catch-all inputs for manual configs.

Reply:
Which makes sense I think - even in a manual config, we show how to sink data into the provided sinks, so the manually-configured data shows up in Better Stack :)